### ISSUE TITLE
[epic] Improve model saving and loading

### ISSUE THREAD

**Issue by mastoffel** [alan-turing-institute/autoemulate#86]
- [ ] #135
    - metadata currently includes `scikit-learn`  and `numpy` versions, which are then checked when the model is loaded. This is quite basic, and we should probably check other things to make sure models work as expected when loaded on other systems. 
- [x] #134
    - Currently, all models are saved using `joblib`, but this might not be the best option for some models. Figure out how to do this better.
- [x] #117

**Comment by mastoffel** [comment]
yes, this is great.Should have a closer look at [Ray](https://www.ray.io/) specifically.

**Comment by kallewesterling** [comment]
@mastoffel This issue would do well with an update following the discussion of implementing Ray as per the comment above.

**Comment by kallewesterling** [comment]
Also, it might actually be an epic issue that we split up into several sub-issues, i.e. one on checking to make sure models work as expected when loaded on other systems, another one on implementing Ray (in lieu of current joblib).

**Comment by mastoffel** [comment]
somewhay outdated, there isn't any metadata anymore
